sa/sa.go

```diff
@@ -227,7 +227,7 @@ func (ssa *SQLStorageAuthority) GetAuthorization(ctx context.Context, id string)
 
 	authz.Challenges, err = ssa.getChallenges(authz.ID)
 	if err != nil {
-		return authz, err
+		return authz, Rollback(tx, err)
```

Comment:
Probably doesn't need to be addressed right now but here we are keeping the transaction open despite have finished doing any transaction work (ssa.getChallenges executes a plain query outside of the transaction). It feels like we should really explicitly commit the transaction as soon as we are done actually working with it instead of holding it open until the end of the method.

Reply:
Yeah. I'd actually like to make this method a non-transaction, but wanted to do the minimally-invasive change for now.